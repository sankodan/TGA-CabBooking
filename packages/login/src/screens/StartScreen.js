import React from 'react'
import { Background, Paragraph, Button, Header, Logo } from '@sankodan/com.cabbooking.components'

export default function StartScreen ({ navigation }) {
  return (
    <Background>
      <Logo />
      <Header></Header>
      <Paragraph>The easiest way to start with your amazing movie booking application.
      </Paragraph>
      <Button
        testID = 'loginScreenButton'
        mode="contained"
        onPress={() => navigation.navigate('LoginScreen')}
      >
        Login
      </Button>
      <Button
        testID = 'RegisterScreenButton'
        mode="outlined"
        onPress={() => navigation.navigate('RegisterScreen')}
      >
        Sign Up
      </Button>
    </Background>
  )
}
