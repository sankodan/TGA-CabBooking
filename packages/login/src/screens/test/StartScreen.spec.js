import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react-native'
import StartScreen from '../StartScreen'
import { shallow } from 'enzyme'

describe('StartScreen', () => {
  const navigate = jest.fn()
  const popToTop = jest.fn()
  const push = jest.fn()
  const navigation = {
    navigate,
    goBack: jest.fn(),
    popToTop,
    push
  }
  describe('clicking loginScreenButton', () => {
    let toJSONObject
    beforeEach(() => {
      const { toJSON } = render(<StartScreen navigation={navigation} />)
      toJSONObject = toJSON
      fireEvent.press(screen.getByTestId('loginScreenButton'))
    })

    it('Dashboard snapshot', () => {
      expect(toJSONObject()).toMatchSnapshot()
    })
  })

  describe('clicking RegisterScreenButton', () => {
    let toJSONObject
    beforeEach(() => {
      const { toJSON } = render(<StartScreen navigation={navigation} />)
      toJSONObject = toJSON
      fireEvent.press(screen.getByTestId('RegisterScreenButton'))
    })

    it('Dashboard snapshot', () => {
      expect(toJSONObject()).toMatchSnapshot()
    })
  })
})
