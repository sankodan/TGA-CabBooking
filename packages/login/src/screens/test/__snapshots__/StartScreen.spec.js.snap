// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreen clicking RegisterScreenButton Dashboard snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="repeat"
    source={
      Object {
        "testUri": "../../../packages/components/assets/background_dot.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "maxWidth": 340,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../packages/components/assets/logo.png",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 200,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#121330",
            "fontSize": 21,
            "fontWeight": "bold",
            "paddingVertical": 12,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "fontSize": 15,
            "lineHeight": 21,
            "marginBottom": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      The easiest way to start with your amazing movie booking application.
    </Text>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginVertical": 10,
          "minWidth": 64,
          "paddingVertical": 2,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="loginScreenButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "lineHeight": 26,
                  },
                ],
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(0, 0, 0, 0.29)",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0.5,
          "elevation": 0,
          "marginVertical": 10,
          "minWidth": 64,
          "paddingVertical": 2,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="RegisterScreenButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "lineHeight": 26,
                  },
                ],
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`StartScreen clicking loginScreenButton Dashboard snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="repeat"
    source={
      Object {
        "testUri": "../../../packages/components/assets/background_dot.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "maxWidth": 340,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../packages/components/assets/logo.png",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 200,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#121330",
            "fontSize": 21,
            "fontWeight": "bold",
            "paddingVertical": 12,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "fontSize": 15,
            "lineHeight": 21,
            "marginBottom": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      The easiest way to start with your amazing movie booking application.
    </Text>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginVertical": 10,
          "minWidth": 64,
          "paddingVertical": 2,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="loginScreenButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "lineHeight": 26,
                  },
                ],
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(0, 0, 0, 0.29)",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0.5,
          "elevation": 0,
          "marginVertical": 10,
          "minWidth": 64,
          "paddingVertical": 2,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="RegisterScreenButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "lineHeight": 26,
                  },
                ],
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
